server/aws-lsp-codewhisperer/src/language-server/telemetryService.test.ts

```diff
@@ -207,4 +209,39 @@ describe('TelemetryService', () => {
         sinon.assert.calledWith(invokeSendTelemetryEventSpy, sinon.match(expectedEvent))
         invokeSendTelemetryEventSpy.restore()
     })
+
+    it('should emit CodeCoverageEvent event', () => {
+        const timestamp = new Date(Date.now())
+        const expectedEvent = {
+            codeCoverageEvent: {
+                customizationArn: 'test-arn',
+                programmingLanguage: { languageName: 'typescript' },
+                acceptedCharacterCount: 123,
+                totalCharacterCount: 456,
+                // timestamp,
+            },
+        }
+        mockCredentialsProvider.setConnectionMetadata({
+            sso: {
+                startUrl: 'idc-start-url',
+            },
+        })
+        telemetryService = new TelemetryService(mockCredentialsProvider, 'bearer', {} as Telemetry, {})
+        const invokeSendTelemetryEventSpy: sinon.SinonSpy = sinon.spy(
+            telemetryService,
+            'invokeSendTelemetryEvent' as any
+        )
+        telemetryService.updateOptOutPreference('OPTIN')
+
+        telemetryService.emitCodeCoverageEvent({
+            languageId: 'typescript',
+            customizationArn: 'test-arn',
+            acceptedCharacterCount: 123,
+            totalCharacterCount: 456,
+        })
+
+        sinon.assert.calledOnce(invokeSendTelemetryEventSpy)
+        sinon.assert.calledWith(invokeSendTelemetryEventSpy, sinon.match(expectedEvent))
```

Comment:
Earlier I was also using that, somehow at times it does not match the exact copy and creates an issue. 

Two things can be changed here - using sinon.stub instead of sinon.spy and second - so the called function with arguments has a timestamp (which cannot be exactly matched if we put timestamp in expectedEvent), we can use something like the one i used in another PR [this](https://github.com/aws/language-servers/pull/555/files#diff-d004eac4f42ad129b7b75ec944d1065dbec5f55d7d996d18d66aede173e58034R207)